src/Core/src/Internal/Config/StateBinder.php

```diff
@@ -58,6 +59,13 @@ public function bind(string $alias, mixed $resolver): void
      */
     public function bindSingleton(string $alias, mixed $resolver): void
     {
+        if ($this->hasInstance($alias)) {
+            throw new DuplicateBindingException(\sprintf(
```

Comment:
You must check the existing in cache. Does the method `hasInstance` do the same?